cmp/grid/GridModel.js

```diff
@@ -732,26 +749,28 @@ export class GridModel {
 
         if (!isEmpty(colIds)) {
             if (this.experimental.useHoistAutosize) {
-                this.hoistAutosize(colIds);
+                await this.hoistAutosize(colIds, options);
             } else {
                 this.agColumnApi?.autoSizeColumns(colIds);
             }
         }
     }
 
     autoSizeColumns(colIds) {
-        console.warn('`GridModel.autoSizeColumns` has been deprecated. Use `GridModel.autosizeColumns()` instead.');
-        this.autosizeColumns(colIds);
+        console.warn('`GridModel.autoSizeColumns` has been deprecated. Use `GridModel.autosizeColumnsAsync()` instead.');
+        this.autosizeColumnsAsync({columns: colIds});
     }
 
     //-----------------------
     // Implementation
     //-----------------------
-    hoistAutosize(colIds) {
-        start(async () => {
-            this.agApi?.showLoadingOverlay();
-            // Wait to allow mask to render before starting potentially compute-intensive autosize.
-            await wait(100);
+    async hoistAutosize(colIds, {showMask}) {
```

Comment:
naming convention: probably should be internalAutosizeAsync at this point, given that the experimental bit is going away.